ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/earliest/EarliestByOffset.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udaf.earliest;
+
+import io.confluent.ksql.function.udaf.Udaf;
+import io.confluent.ksql.function.udaf.UdafDescription;
+import io.confluent.ksql.function.udaf.UdafFactory;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+@UdafDescription(
+    name = "EARLIEST_BY_OFFSET",
+    description = EarliestByOffset.DESCRIPTION
+)
+public final class EarliestByOffset {
+  static final String DESCRIPTION =
+      "This function returns the oldest value for the column, computed by offset.";
+
+  private EarliestByOffset() {
+  }
+
+  static final String SEQ_FIELD = "SEQ";
+  static final String VAL_FIELD = "VAL";
+
+  public static final Schema STRUCT_INTEGER = SchemaBuilder.struct().optional()
```

Comment:
All these constants look the same as the ones in LatestbyOffset. It would be nice to move them to another shared file instead of duplicating them.